fusor-ember-cli/app/controllers/configure-environment.js

```diff
@@ -45,10 +45,15 @@ export default Ember.Controller.extend(ConfigureEnvironmentMixin, NeedsDeploymen
     createEnvironment(fields_env) {
       var self = this;
       this.set('showAlertMessage', false);
+      this.set('errorMsg', null);
+      this.get('deploymentController').set('errorMsg', null);
 
-      var nameAlreadyExists =  self.get('lifecycleEnvironments').findBy('name', fields_env.name);
+      var nameAlreadyExists =  this.get('lifecycleEnvironments').findBy('name', fields_env.name);
       if (nameAlreadyExists) {
-        self.get('deploymentController').set('errorMsg', fields_env.name + ' is not a unique name. Environment not saved.');
+        let errorMsg = fields_env.name + ' is not a unique name. Environment not saved.';
```

Comment:
previously, there was not `return` so the function continued and showed a different error message ```'error saving environment' + error```, but there's no need to hit the back server in there's a duplicate, so this is the improvement.